scripts/paged_speculative_inference.py

```diff
@@ -0,0 +1,234 @@
+import argparse
+import itertools
+import os
+import time
+
+import torch
+import torch._inductor.config
+from fms.models import get_model
+from fms.utils import generation, tokenizers
+from torch import distributed as dist
+
+import fms_extras.models.paged_llama
+from fms_extras.utils.generation import paged_generate
```

Comment:
other than using paged_generate, is everything else in this file copy/paste? it would be nice to improve code reuse for these kinds of scripts, maybe by modifying the ones in fms.

Reply:
This was adapted from the one in `fms`, but includes a warmup, some extra metrics, initializes the kv-cache, as well as calls different functions for generate (including speculator in later PR). Might make sense at some point to add some generic script for inference in `fms` that we could extend with our own custom functions to create other inference scripts. For now, I think this should be kept separate as there is enough differences.